backend/service/aws/dynamodb_test.go

```diff
@@ -110,11 +124,133 @@ func TestDescribeTableWithGsiValid(t *testing.T) {
+	assert.Equal(t, ds.MaxScaleFactor, SafeScaleFactor, "scale factor default set")
+	assert.False(t, ds.EnableOverride)
+}
+
+func TestGetScalingLimitsCustom(t *testing.T) {
+	cfg := &awsv1.Config{
+		Regions: regions,
+		ClientConfig: &awsv1.ClientConfig{
+			Retries: 10,
+		},
+		DynamodbConfig: &awsv1.DynamodbConfig{
+			ScalingLimits: &awsv1.ScalingLimits{
+				MaxReadCapacityUnits:  1000,
+				MaxWriteCapacityUnits: 2000,
+				MaxScaleFactor:        4.0,
+				EnableOverride:        false,
+			},
+		},
+	}
+
+	ds := getScalingLimits(cfg)
+
+	assert.Equal(t, ds.MaxReadCapacityUnits, cfg.DynamodbConfig.ScalingLimits.MaxReadCapacityUnits, "RCU set")
+	assert.Equal(t, ds.MaxWriteCapacityUnits, cfg.DynamodbConfig.ScalingLimits.MaxWriteCapacityUnits, "WCU set")
+	assert.Equal(t, ds.MaxScaleFactor, cfg.DynamodbConfig.ScalingLimits.MaxScaleFactor, "scale factor default set")
+	assert.False(t, ds.EnableOverride)
+}
+func TestUpdateTableCapacityWithDefaultLimits(t *testing.T) {
+	tests := []struct {
+		name     string
+		inputRCU int64
+		inputWCU int64
+		want     string
+	}{
+		{"rcu above max", 100000, 250, "rpc error: code = FailedPrecondition desc = Target read capacity exceeds maximum allowed limits [40000]"},
+		{"wcu above max", 100, 100000, "rpc error: code = FailedPrecondition desc = Target write capacity exceeds maximum allowed limits [40000]"},
+		{"rcu lower than current", 1, 1500, "rpc error: code = FailedPrecondition desc = Target read capacity [1] is lower than current capacity [100]"},
+		{"wcu lower than current", 1500, 1, "rpc error: code = FailedPrecondition desc = Target write capacity [1] is lower than current capacity [200]"},
+		{"rcu change scale too high", 400, 200, "rpc error: code = FailedPrecondition desc = Target read capacity exceeds the scale limit of [2.0]x current capacity"},
+		{"wcu change scale too high", 100, 600, "rpc error: code = FailedPrecondition desc = Target write capacity exceeds the scale limit of [2.0]x current capacity"},
+	}
+
+	m := &mockDynamodb{
+		table: testDynamodbTable,
+	}
+
+	ds := getScalingLimits(cfg)
+
+	d := &awsv1.DynamodbConfig{
+		ScalingLimits: &awsv1.ScalingLimits{
+			MaxReadCapacityUnits:  ds.MaxReadCapacityUnits,
+			MaxWriteCapacityUnits: ds.MaxWriteCapacityUnits,
+			MaxScaleFactor:        ds.MaxScaleFactor,
+			EnableOverride:        ds.EnableOverride,
+		},
+	}
+
+	c := &client{
+		log:     zaptest.NewLogger(t),
+		clients: map[string]*regionalClient{"us-east-1": {region: "us-east-1", dynamodbCfg: d, dynamodb: m}},
+	}
+
+	for _, tt := range tests {
+		tt := tt // capture range variable
+		t.Run(tt.name, func(t *testing.T) {
+			got := c.UpdateTableCapacity(context.Background(), "us-east-1", "test-gsi-table", tt.inputRCU, tt.inputWCU)
+			if got.Error() != tt.want {
+				t.Errorf("\nWant error msg: %s\nGot error msg: %s", tt.want, got)
+			}
+		})
+	}
+}
+
+func TestUpdateTableCapacityWithCustomLimits(t *testing.T) {
```

Comment:
do we test the success case?

Reply:
Good catch, I'll add tests for success cases